test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ViewFeatures/DefaultHtmlGeneratorTest.cs

```diff
@@ -190,6 +191,103 @@ public void GenerateTextArea_WithNullExpression_WithNameAttribute_DoesNotThrow()
             Assert.Equal(expected, attribute.Value);
         }
 
+        [Theory]
+        [MemberData(nameof(GetMaxLengthAttributeRenderingTestsData))]
+        public void Generate_RendersMaxLengthAttribute_AsRequested(
+            string expression,
+            int expectedValue,
+            Action<Dictionary<string, object>> customizeHtmlAttributes,
+            bool maxLengthShouldBeRendered,
+            Func<IHtmlGenerator, ViewContext, ModelExplorer, Dictionary<string, object>, string, TagBuilder> getTagBuilder)
+        {
+            // Arrange
+            var metadataProvider = new TestModelMetadataProvider();
+            var htmlGenerator = GetGenerator(metadataProvider);
+            var viewContext = GetViewContext<ModelWithMaxLengthMetadata>(model: null, metadataProvider: metadataProvider, new HtmlHelperOptions { MaxLengthAttributeRenderingEnabled = true });
```

Comment:
Why is it necessary to pass `HtmlHelperOptions` into `GetViewContext(...)`? The generator has already been instantiated with `MaxLengthAttributeRenderingEnabled = true`.

Reply:
Forgot to remove the first addition I made. Will remove